<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <!-- Joint State Publisher for reading joint positions -->
    <gazebo>
        <plugin name="joint_state_publisher" filename="libgazebo_ros_joint_state_publisher.so">
            <update_rate>100</update_rate>
            <joint_name>joint_1</joint_name>
            <joint_name>joint_2</joint_name>
            <joint_name>joint_3</joint_name>
        </plugin>
    </gazebo>

    <!-- Individual joint control plugins -->
    <gazebo reference="joint_1">
        <plugin name="joint_1_controller" filename="libgazebo_ros_joint_pose_trajectory.so">
            <update_rate>100</update_rate>
        </plugin>
    </gazebo>

    <gazebo reference="joint_2">
        <plugin name="joint_2_controller" filename="libgazebo_ros_joint_pose_trajectory.so">
            <update_rate>100</update_rate>
        </plugin>
    </gazebo>

    <gazebo reference="joint_3">
        <plugin name="joint_3_controller" filename="libgazebo_ros_joint_pose_trajectory.so">
            <update_rate>100</update_rate>
        </plugin>
    </gazebo>

</robot>